### PR TITLE
Upgrading to the new integer protocols

### PR DESCRIPTION
**Pull request by moiseev**

### CODE REVIEW

File: Foundation/CGFloat.swift [hunk 1]
```diff
@@ -279,33 +292,28 @@ extension CGFloat : BinaryFloatingPoint {
...
-
-    @_transparent
     public mutating func negate() {
         native.negate()
     }
 
     @_transparent
-    public mutating func add(_ other: CGFloat) {
-        native.add(other.native)
+    public static func +=(_ lhs: inout CGFloat, _ rhs: CGFloat) {
```
Comment (parkera) [inline Foundation/CGFloat.swift]: Is this a source-compatible change?

Comment (moiseev) [inline Foundation/CGFloat.swift]: Technically not, but these methods were there only to implement operators on the protocol, which is no longer required. In any case these missing methods can now be added for source compatibility in the protocol. Thanks for looking at this!

Comment (moiseev) [inline Foundation/CGFloat.swift]: Added the missing methods here:  https://github.com/apple/swift/pull/3796/commits/192c11217fe0c1a1821004b34434ffe523875707

**Comment by moiseev** [comment]
@parkera I've just merged the new integers into master. Is it OK for me to merge this one?

**Comment by slavapestov** [comment]
This is blocking PR testing on Linux so I'm going to merge it.

**Pull request merged**